### ISSUE TITLE
Minecraft Development Plugin - Recaf Edition

### ISSUE THREAD

**Issue by Amejonah1200** [Col-E/Recaf#667]
Because of popular demand and usage of Recaf, it would be advised to create as specific plugin handling those features:
- Ability to download mappings of MC.
- Presets of resolution of dependencies using #666 for Forge, Fabric, MC, Paper, Spigot, etc.

**Comment by pvpb0t** [comment]
Didnt [Filip](https://github.com/1fxe) already do almost this in his Recaf4Forge?

**Comment by HoratioGamer** [comment]
This would be very useful.   What might be a first step is, if Recaf could say give examples in the documentation of how to load the yarn/mappings: 

https://github.com/FabricMC/yarn/tree/1.20.5-pre2/mappings

that would be really helpful.   What type of mappings are the yarn mappings?  Simple?  Enigma?  Proguard? SRG, TSRG?  Tiny?  Tiny V2?  JADX?   Yarn does not say, or does not use the same terminology as Recaf.  

If Recaf already has the ability to read yarn/mappings then if it were simply explained in a page of documentation, that would be a large part of the problem solved.   

If Recaf does not yet have the ability to read yarn/mappings, 

a) if there are simple work-arounds, like, concatenate all the files from the yarn/mappings tree into a single text file, and load that using the mappings tab of Recaf, with the Simple/Enigma..../JADX setting that is needed then that would be useful to explain in docs.   

b) if there is a simple utility to convert yarn/mappings to a format Recaf can use, then it would be really useful if the docs could provide links and explanations on how to do that.   

c) If there were documentation about the mappings tab of Recaf, explaining the format, and giving examples for each type of mapping then someone reasonably skilled in Perl or TCL could write a utility to read yarn/mappings and write them out in a format that Recaf can read.   

In summary, this might be just a documentation problem, or maybe documentation could go a long way, with available utilities, or ones we can write, to helping us solve the problem of using Recaf for Minecraft Mod Programming.   

Thank you for your time.

**Comment by Col-E** [comment]
d24545ba7afe74e85dab9056074611796e20bf52 - Some progress for built-in Minecraft content support
